### PR TITLE
34 dev energy density unit extension in ontouom

### PR DESCRIPTION
**Pull request by YiKaiTsai**

### CODE REVIEW

**Review by markushofmeister — CHANGES_REQUESTED** [review]
Just looking at the units needed for OntoHeatNetwork: Is `EUR/t` already included in `OM` or did we just miss it?

**Comment by markushofmeister** [comment]
@msff2: If I remember correctly, the newly created concepts would need to be added to the overall `om-2.0.rdf` file on `https://github.com/cambridge-cares/OM` (as well as Claudius - to ensure resolvability), right? If that's still the case:
1) Could you please guide Yi Kai through that (I am very sorry, but I doubt to remember all steps properly)
2) Does the changed base IRI of the additional concepts cause issues?

**Review by sm453 — CHANGES_REQUESTED** [review]
The unit "kilowattHourPerKilowatt" appears to be labelled "kilowatt hour per square metre". Please fix this.

File: JPS_Ontology/ontology/ontouom/kilowattHourPerKilowatt [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+  <!-- Kilowatt Hour Per Kilowatt Ontology -->
+
+  <om:UnitDivision rdf:about="https://www.theworldavatar.com/kg/ontouom/kilowattHourPerKilowatt">
+    <rdfs:label xml:lang="en">kilowatt hour per square metre</rdfs:label>
+    <om:symbol>kWh/kW</om:symbol>
+    <om:alternativeSymbol>kWh kW-1</om:alternativeSymbol>
+    <om:alternativeSymbol>kWh·kW-1</om:alternativeSymbol>
+    <om:hasNumerator rdf:resource="&om;kilowattHour"/>
+    <om:hasDenominator rdf:resource="&om;kilowatt"/>
+    <om:hasDimension rdf:resource="&om;time-Dimension"/>
```
[the remark this answers is not in the document]
Comment (YiKaiTsai) [reply JPS_Ontology/ontology/ontouom/kilowattHourPerKilowatt:38]: This is for Solarkataster, meant for specific yield.

File: JPS_Ontology/ontology/ontouom/euroPerHour [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  <!-- Euro Per Hour Ontology -->
+  
+  <om:UnitDivision rdf:about="https://www.theworldavatar.com/kg/ontouom/euroPerHour">
+     <rdfs:label xml:lang="en">euro per hour</rdfs:label>
+     <om:symbol>EUR/h</om:symbol>
+     <om:alternativeSymbol>EUR h-1</om:alternativeSymbol>
+     <om:alternativeSymbol>EUR·h-1</om:alternativeSymbol>
+     <om:hasNumerator rdf:resource="&om;euro"/>
+     <om:hasDenominator rdf:resource="&om;hour"/>
+   </om:UnitDivision>
```
[the remark this answers is not in the document]
Comment (YiKaiTsai) [reply JPS_Ontology/ontology/ontouom/euroPerHour:38]: > Just looking at the units needed for OntoHeatNetwork: Is `EUR/t` already included in `OM` or did we just miss it?

EUR/t is already in OM as euroPerTonne.

**Review by markushofmeister — APPROVED** [review]

**Review by msff2 — APPROVED** [review]

File: JPS_Ontology/ontology/ontouom/euroPerHour (hunk 2 above)
[the remark this answers is not in the document]
Comment (msff2) [reply JPS_Ontology/ontology/ontouom/euroPerHour:38]: > > Currently we define all concepts using this unit solely as `Quantity` (e.g. revenue in time interval, cost in time interval); hence, there is no specific `subQuantity` and `UnitDimension` defined. Is this fine for now or shall that be amended? Same applies for `euroPerMegawattHour`
> 
> @sm453 @msff2 Is this fine for now or shall we create more specific `Sub-Quantities`? Looking at `https://raw.githubusercontent.com/cambridge-cares/OM/master/om-2.0.rdf`, I can see that there is already a `AmountOfMoneyPerArea` concept specified as subclass of `Quantity`. There are also the required `Unit` and `Dimension` concepts included. Following this example, it should be easy to create the corresponding concepts for `AmountOfMoneyPerTimeInterval` and `AmountOfMoneyPerEnergy`.

Seb wants us to put all Units of Measure ontology extensions in our OM fork, as this is an ideal approach to follow. I think it won't be a bad idea to ask [YiKaiTsai](https://github.com/YiKaiTsai) to create a more specific quantity for each new unit wherever required by analysing the Cost, AmountOfMoney and SpecificAmountOfMoney subclasses of Quantity and their relationships. We can postpone the migration to the OM fork until [YiKaiTsai](https://github.com/YiKaiTsai) completes this task.

**Comment by msff2** [comment]
> 



> @msff2: If I remember correctly, the newly created concepts would need to be added to the overall `om-2.0.rdf` file on `https://github.com/cambridge-cares/OM` (as well as Claudius - to ensure resolvability), right? If that's still the case:
> 
> 1. Could you please guide Yi Kai through that (I am very sorry, but I doubt to remember all steps properly)
> 2. Does the changed base IRI of the additional concepts cause issues?

Soon, we will include everything in om-2.0.rdf. Yes, [YiKaiTsai](https://github.com/YiKaiTsai) should put newly created classes/concepts in that file. Regarding my help or guidance, please read the previous comment on analysing existing subclasses of Quantity.

**Review by sm453 — CHANGES_REQUESTED** [review]
I believe in JPS_Ontology/ontology/ontouom/nanogramPerCubicMetre, a hasDimension node has been accidentally replaced with a URL string.

In JPS_Ontology/ontology/ontouom/EnergyPerArea, there are three nodes defining different aspects of the same class. Wouldn't it be preferable to merge these into one?

How come JPS_Ontology/ontology/ontouom/MassPerEnergy is treated so differently from others, e.g. JPS_Ontology/ontology/ontouom/HeatFluxDensity? Shouldn't it also have a value and a unit, etc.?

**Review by msff2 — APPROVED** [review]

**Comment by msff2** [comment]
> I believe in JPS_Ontology/ontology/ontouom/nanogramPerCubicMetre, a hasDimension node has been accidentally replaced with a URL string.

I agree with Seb's doubt and this needs to be fixed.

**Comment by msff2** [comment]
> In JPS_Ontology/ontology/ontouom/EnergyPerArea, there are three nodes defining different aspects of the same class. Wouldn't it be preferable to merge these into one?

Yes, these are different aspects of the same class and we can merge them. I think Yi has done it by following OM.

**Comment by msff2** [comment]
> How come JPS_Ontology/ontology/ontouom/MassPerEnergy is treated so differently from others, e.g. JPS_Ontology/ontology/ontouom/HeatFluxDensity? Shouldn't it also have a value and a unit, etc.?

Yes, I agree with Seb that, at the moment, you created a partial representation of MassPerEnergy. I think in doing this, you looked at the partial model like the following for Length:

  <owl:Class rdf:about="&om;Length">
    <rdfs:subClassOf>
      <owl:Restriction>
        <owl:onProperty rdf:resource="&om;hasDimension"/>
        <owl:hasValue rdf:resource="&om;length-Dimension"/>
      </owl:Restriction>
    </rdfs:subClassOf>
  </owl:Class>

But you didn't notice that the same class Length also has the following representation in the same om-2.0.rdf file.

<owl:Class rdf:about="&om;Length">
    <rdfs:subClassOf>
      <owl:Restriction>
        <owl:onProperty rdf:resource="&om;hasValue"/>
        <owl:allValuesFrom>
          <owl:Restriction>
            <owl:onProperty rdf:resource="&om;hasUnit"/>
            <owl:allValuesFrom>
              <owl:Class rdf:about="&om;LengthUnit"/>
            </owl:allValuesFrom>
          </owl:Restriction>	
        </owl:allValuesFrom>
      </owl:Restriction>     
    </rdfs:subClassOf>     
  </owl:Class>

So please represent all classes for units similarly.

File: JPS_Ontology/ontology/ontouom/MassPerEnergy [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+    xmlns:rdfs="&rdfs;"
+    xmlns:om="&om;"
+    xmlns="&om;"
+    xml:base="&om;"
+  >
+
+  <owl:Ontology rdf:about="">
+    <rdfs:label xml:lang="en">Ontology of units of Measure (OM)</rdfs:label>
+  </owl:Ontology>
+  </owl:Ontology>
```
Comment (msff2) [inline JPS_Ontology/ontology/ontouom/MassPerEnergy:28]: Remove </owl:Ontology> from line 28.

File: JPS_Ontology/ontology/ontouom/HeatFluxDensity [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    xmlns:rdfs="&rdfs;"
+    xmlns:om="&om;"
+    xmlns="&om;"
+    xml:base="&om;"
+  >
+
+  <owl:Ontology rdf:about="">
+    <rdfs:label xml:lang="en">Ontology of units of Measure (OM)</rdfs:label>
+  </owl:Ontology>
+  </owl:Ontology>
```
Comment (msff2) [inline JPS_Ontology/ontology/ontouom/HeatFluxDensity:28]: Remove </owl:Ontology> from line 28.

**Comment by YiKaiTsai** [comment]
> How come JPS_Ontology/ontology/ontouom/MassPerEnergy is treated so differently from others, e.g. JPS_Ontology/ontology/ontouom/HeatFluxDensity? Shouldn't it also have a value and a unit, etc.?

MassPerEnergy was already defined in the original OM, but without a defined dimension. The massPerEnergy-Dimension was defined by us, and my intend was to append it to the MassPerEnergy definition as its defined dimension. I left out the MassPerEnergy value and unit parts, since they were already in OM. I could copy and paste the value and unit parts from OM into our file, if needed.

**Review by sm453 — APPROVED** [review]
All good. No need for further action regarding MassPerEnergy.

**Pull request merged**